frontend/src/components/task/task.js

```diff
@@ -688,7 +688,14 @@ class Task extends Component {
     const isAssignOwner = () => {
       return taskOwner() || isCurrentUserAssigned()
     }
-
+    // Error handling when task does not exist
+    if (task.completed && !task.values) {
+      this.props.history.push('/404')
+      return null
+    }
+    if (!task.completed) {
+      return (<div />)
+    }
```

Comment:
@ShubhamCanMakeCommit We should maybe handle this on actions when we try to fetch a task? And the server should return `404` and we detect the response status to actually redirect to 404 page?

Reply:
But we can't call `this.props.history.push('/404')` inside [taskActions.js- syncTask()](https://github.com/worknenjoy/gitpay/blob/master/frontend/src/actions/taskActions.js#L532)
And I don't know how would I implement the other way. Maybe you can show the way on how to implement. Just a hint maybe.

Reply:
Yeah, let's keep for now this solution on the back-end and further look into a more generic solution for all 404 and 500